src/DynamoCoreWpf/ViewModels/Search/SearchViewModel.cs

```diff
@@ -86,10 +94,47 @@ public string SearchText
             set
             {
                 searchText = value;
-                OnSearchTextChanged(this, EventArgs.Empty);
+                // The searchText is set multiple times before the control becomes visible and interactable.
+                // To prevent any debounces from triggering at some unexpected point before or after the control
+                // becomes visible, this flag is only set once the searchText value is set by the user
+                // (unless it somehow gets set somewhere else)
+                searchInitialized = searchInitialized || !string.IsNullOrEmpty(searchText);
+
+                if (searchDebouncer == null && dynamoViewModel?.UIDispatcher is Dispatcher dispatcher)
+                {
+                    // Dipatcher.CurrentDispatcher has a chance of returning the wrong dispatcher, so we need to
+                    // use the dynamoViewModel.UIDispatcher. The UIDispatcher is set after the SearchViewModel
+                    // constructor runs, otherwise the debouncer could be initalized there
+                    searchDebouncer = new ActionDebouncer(dispatcher);
```

Comment:
why not initialize this class in the SearchViewModel constructor ?
You could also pass the logger to the ActionDebouncer instance and do the try catch inside there (I am assuming this class is supposed to be reused through out dynamo)